openedx_learning/apps/authoring/publishing/api.py

```diff
@@ -513,3 +558,419 @@ def filter_publishable_entities(
+    """
+    assert issubclass(container_version_model, ContainerVersion)
+    with atomic(savepoint=False):  # Make sure this will happen atomically but we don't need to create a new savepoint.
+        publishable_entity_version = create_publishable_entity_version(
+            container.publishable_entity_id,
+            version_num=version_num,
+            title=title,
+            created=created,
+            created_by=created_by,
+        )
+        container_version = container_version_model.objects.create(
+            publishable_entity_version=publishable_entity_version,
+            container_id=container.pk,
+            entity_list=entity_list,
+        )
+
+    return container_version
+
+
+def create_container_version(
+    container_id: int,
+    version_num: int,
+    *,
+    title: str,
+    publishable_entities_pks: list[int],
+    entity_version_pks: list[int | None] | None,
+    created: datetime,
+    created_by: int | None,
+    container_version_model: type[ContainerVersionModel] = ContainerVersion,  # type: ignore[assignment]
+) -> ContainerVersionModel:
+    """
+    [ 🛑 UNSTABLE ]
+    Create a new container version.
+
+    Args:
+        container_id: The ID of the container that the version belongs to.
+        version_num: The version number of the container.
+        title: The title of the container.
+        publishable_entities_pks: The IDs of the members of the container.
+        entity_version_pks: The IDs of the versions to pin to, if pinning is desired.
+        created: The date and time the container version was created.
+        created_by: The ID of the user who created the container version.
+        container_version_model: The subclass of ContainerVersion to use, if applicable.
+
+    Returns:
+        The newly created container version.
+    """
+    assert title is not None
+    assert publishable_entities_pks is not None
+
+    with atomic(savepoint=False):
+        container = Container.objects.select_related("publishable_entity").get(pk=container_id)
+        entity = container.publishable_entity
+        if entity_version_pks is None:
+            entity_version_pks = [None] * len(publishable_entities_pks)
+        entity_list = create_entity_list_with_rows(
+            entity_pks=publishable_entities_pks,
+            entity_version_pks=entity_version_pks,
+            learning_package_id=entity.learning_package_id,
+        )
+        container_version = _create_container_version(
+            container,
+            version_num,
+            title=title,
+            entity_list=entity_list,
+            created=created,
+            created_by=created_by,
+            container_version_model=container_version_model,
+        )
+
+    return container_version
+
+
+def create_next_container_version(
+    container_pk: int,
+    *,
+    title: str | None,
+    publishable_entities_pks: list[int] | None,
+    entity_version_pks: list[int | None] | None,
```

Comment:
Nit: I think it's slightly more natural to send tuples than have these separate, but again, non-blocking commentary.

Reply:
I would like to do that, and I'd like to also introduce some sort of verification that the entity version objects actually match the corresponding entity objects, but I think I'll leave that to another PR or get someone from my team to follow up.